### ISSUE TITLE
Wrong marker position on smaller zoom numbers

### ISSUE THREAD

**Issue by virtualize** [DantSu/php-osm-static-api#5]
Thank you very much for this fine library.

It seems like a marker it getting shifted up the smaller the zoom is, i.e. for zoom 12 the marker position seems good. If I decrease the zoom stepwise to smaller numbers, the marker will be positioned above (and more left) of the given coordinates.

In the following images the zoom is decreased from 12 to 8, 6 and 4.
At zoom 4 the marker is almost at germanys norther border, but the coordinates point to a place way south.

![db7b1e5f-a6db-4942-9ee7-f1f59f245842](https://user-images.githubusercontent.com/220102/161311543-1b249db5-58d3-4c66-93c5-2cf21d3f2978.png)
![efddea52-d2e2-461e-849d-865ff37e0303](https://user-images.githubusercontent.com/220102/161311597-a207180b-128f-438d-bbbc-8c4745641266.png)
![f68da997-483f-48b0-9767-d4921d17a078](https://user-images.githubusercontent.com/220102/161311652-591267ff-cc32-4304-8316-ab5b23f8c71e.png)
![4eac9f48-f4e7-4e88-920f-a14867c77152](https://user-images.githubusercontent.com/220102/161311669-f7eb1ce6-66d7-4e0b-a86f-4e9f0b98088a.png)

My setup is rather simple
```php
$tempFilePath = tempnam(sys_get_temp_dir(), 'osm');
$markerPath = $this->parameterBag->get('kernel.project_dir').'/assets/images/marker-black.png';

$markers = (new Markers($markerPath))
    ->setAnchor(Markers::ANCHOR_CENTER, Markers::ANCHOR_BOTTOM)
    ->addMarker(new LatLng($lat, $lng));

$map = (new OpenStreetMap(new LatLng($lat, $lng), $zoom, 640, 640))
    ->addMarkers($markers);

$map->getImage()->savePNG($tempFilePath);
```

I had a quick look at [Markers::draw()](https://github.com/DantSu/php-osm-static-api/blob/80d25c35ea4f2fb6ac209eac0db60b76d4ecabf8/src/Markers.php#L79) but could not find the problem.

Edit: The marker anchor position does not seem to have anything to do with the misplacement.

**Comment by virtualize** [comment]
Thanks for your fast response. v0.1.2 is a bit better. Meaning the marker is still shifted above, but not that much as before. Then I've noticed that the marker misplacement highly depends on the image size. (I double checked and this was also the case in v0.1.1.)

I did all tests with zoom level 6. In v0.1.2. its almost perfect with image size 512x512.
If I increase the size to 1024x1024 or even 2048x2048 the marker will be heavily shifted above, see screenshots.

512x512
![512](https://user-images.githubusercontent.com/220102/161641781-2e1a0b7c-d8fa-4243-916c-22feb77ddfdd.jpeg)

1024x1024
![1024](https://user-images.githubusercontent.com/220102/161641809-cc5399f4-2407-4394-9916-117178523ae4.jpeg)

2048x2048
![2048](https://user-images.githubusercontent.com/220102/161641837-eb190e1f-d532-42c5-9276-43326d61cb28.jpeg)

So to sum up: The error gets bigger the smaller the zoom level and the bigger the image size is.

By now I use 512x512 and zoom level 6, so the marker placement is good enough for me now. 
I will not ask for more, considering you're already sharing this package openly :) So thank you very much.

**Comment by DantSu** [comment]
I will try another way to solve this. I think that come from php math precision that is not good enougth.

**Comment by virtualize** [comment]
Looks pretty good and works as expected. Thanks a lot.

I you want me to nitpick then I have these 2 extreme examples, where it's still misplacing the marker:
Size: 512x512, zoom 4 and zoom 2, original marker placement was center of Paris:

![4](https://user-images.githubusercontent.com/220102/162162488-5667e092-5515-41c1-a314-bc53c7e3d155.jpg)
![2](https://user-images.githubusercontent.com/220102/162162515-c29c012f-0599-4eaa-8648-ca54ec0bf4f3.jpg)

But for me these use cases are not relevant and I would close the ticket if you agree.

**Comment by DantSu** [comment]
Very low zoom means a big latitude difference between top and bottom of the tile. I think this is the problem. I close this issue but I will look for solve this problem in the future.

**Comment by DantSu** [comment]
Thanks for your feebback *

**Comment by virtualize** [comment]
Thanks a lot! 👍
